feeds/backend.php

```diff
@@ -4,22 +4,43 @@
 include_once($relPath.'pg.inc');
 
 $content = get_enumerated_param($_GET, 'content', 'posted', ['posted', 'postprocessing', 'proofing', 'smoothreading']); // Which feed the user wants
-// Time in seconds for how often the feeds get refreshed
-// Disable delay if we are on the test server
-$refreshDelay = $testing ? 0 : 30 * 60;
-$refreshAge = time() - $refreshDelay; // How long ago $refreshDelay was in UNIX time
-$limit = 20; // Number of rows we query from the table, number of items in RSS feed
+$rssfeed = generate_rss_feed($content, $site_name, $code_url, $charset, $site_manager_email_addr);
 
-// Determine if we should display a 0.91 compliant RSS feed or our own feed
-$intlang = get_desired_language();
-if (isset($_GET['type'])) {
-    $xmlfile = "$xmlfeeds_dir/{$content}_rss.$intlang.xml";
-} else {
-    $xmlfile = "$xmlfeeds_dir/{$content}.$intlang.xml";
-}
+// Let the browser cache it for $cache_duration seconds
+$cache_duration = 30 * 60;
+$now = time();
+header("Content-Type: text/xml; charset=$charset");
+header("Expires: " . gmdate("D, d M Y H:i:s", $now + $cache_duration) . " GMT");
+header("Last-Modified: " . gmdate("D, d M Y H:i:s", $now) . " GMT");
+header("Cache-Control: max-age=$cache_duration, public, must-revalidate");
+
+echo $rssfeed;
+
+/**
+ * Generate an RSS feed for a category of eBooks
+ *
+ * @param string $content
+ *   The category for the feed: posted, postprocessing, proofing, or smoothreading
+ *
+ * @param string $site_name
+ *   The name of the site serving the feed
+ *
+ * @param string $code_url
+ *   The base URL for the site
+ *
+ * @param string $charset
```

Comment:
`$charset` is always going to be `UTF-8` since the conversion to Unicode in 2020. I have a branch ready to remove this global from the codebase but was waiting on this PR to land first so I didn't generate conflicts. Lets drop the variable here and just put `UTF-8` in the XML header as a literal.

Reply:
Done. I changed it in the HTTP response header too.